pkg/subscriptions/subscriptions.js

```diff
@@ -0,0 +1,276 @@
+                    status = status + '(' + segment_info['Status Details'] + ')';
+
+                /* convert text output to mustache template variables */
+                ret.push({
+                    'product_name': segment_info['Product Name'],
+                    'product_id': segment_info['Product ID'],
+                    'version': segment_info['Version'],
+                    'arch': segment_info['Arch'],
+                    'status': status,
+                    'starts': segment_info['Starts'],
+                    'ends': segment_info['Ends'],
+                });
+            }
+            return ret;
+        }
+
+        function get_subscriptions_info() {
+            cockpit.spawn(['subscription-manager', 'list'],
+                    { directory: '/', superuser: true })
+                .done(function(output) {
+                    is_updating.hide();
+                    var subscriptions = parse_multiple_subscriptions(output);
+                    if ((subscriptions.length > 0)  && (output.indexOf('Subscribed') > -1)) {
+                        render_subscriptions(subscriptions);
+                        system_unregistered.hide();
+                    }
+                    else {
+                        /* no valid subscription: hide output and present user with the option to register system */
+                        subscription_info.hide();
+                        system_unregistered.show();
+                    }
+
+                    // TODO: get registration info as well
+                })
+                .fail(function(ex) {
+                    is_updating.hide();
+                    if (ex.problem != 'not-authorized')
+                        console.warn("Subscriptions: couldn't get info: " + ex);
+                });
+        }
+
+        /* request update via DBus */
+        function update_subscriptions() {
+            var call = service.call(
+                '/EntitlementStatus',
+                'com.redhat.SubscriptionManager.EntitlementStatus',
+                'check_status',
+                []);
+
+            call.fail(function(ex) {
+                    console.warn("Subscriptions: couldn't request update: " + ex);
+                });
+        }
+
+        /* we want to get notified if subscription status of the system changes */
+        service.subscribe(
+            { path: '/EntitlementStatus',
+              interface: 'com.redhat.SubscriptionManager.EntitlementStatus',
+              member: 'entitlement_status_changed'
+            },
+            function(path, dbus_interface, signal, args) {
+                /*
+                 * status has changed, now get actual status via command line
+                 * since older versions of subscription-manager don't deliver this via DBus
+                 * note: subscription-manager needs superuser privileges
+                 */
+                get_subscriptions_info();
```

Comment:
You need to do some sort of rate limiting, or you could easily get into a situation where you are invoking ```subscription-manager list``` 10 times all at once.